packages/eas-cli/src/credentials/ios/actions/SetupBuildCredentials.ts

```diff
@@ -46,7 +48,10 @@ export class SetupBuildCredentials implements Action<IosAppBuildCredentials> {
           bundleIdentifier: app.bundleIdentifier,
           projectName: app.projectName,
         },
-        { enablePushNotifications: true }
+        {
+          // TODO: Get correct workflow
+          entitlements: await resolveEntitlementsJsonAsync(ctx.projectDir, Workflow.GENERIC),
```

Comment:
this code should not be aware of workflow because it depended on the way it's executed, users potentially can have build profiles with both generic managed workflow.

This class can be executed when building or from credentials manager. I suggest
- reading running resolveEntitlementsJsonAsync in here https://github.com/expo/eas-cli/blob/5e5badf11f2699a4e855cc15a1105547f72bf9aa/packages/eas-cli/src/build/ios/credentials.ts#L22 and passing entitlements to this class
- assume it's managed here https://github.com/expo/eas-cli/blob/5e5badf11f2699a4e855cc15a1105547f72bf9aa/packages/eas-cli/src/credentials/manager/ManageIos.ts#L226